File: java/Observer/complete/Observable.java
import java.util.ArrayList;

/*
the "subject" or "observable" class, state changes in this class are sent to
the observer
*/
public class Observable {

  //initialize list of observer and state
  public ArrayList<Observer> observers = new ArrayList<Observer>();
  public String state = "";

  //add an observer and give it Observable's current state
  public void add_observer(Observer obs){
    observers.add(obs);
    obs.notify(this.state);
  }

  //remove an observer from our list
  public void remove_observer(Observer obs){
    this.observers.remove(obs);
  }

  //notify all observers about our current state
  public void notify_observers(){
    for(Observer obs : this.observers){
      obs.notify(this.state);
    }
  }

  //set the current state and notify our Observers of the change
  public void set_state(String state){
    this.state = state;
    this.notify_observers();
  }
}

File: python/Observer/complete/Observable.py
from Observer import Observer

class Observable(object):
    '''
    the 'subject' or 'observable' class, state changes in this class are sent to the
    observers
    '''

    def __init__(self):
        #initialize our list of observers
        self.observers = []
        self.state = "ready"

    def add_observer(self, obs):
        #add an observer to our list of observers and notify it
        self.observers.append(obs)
        obs.notify(self.state)

    def remove_observer(self, obs):
        #remove an observer from our list
        self.observers = list(filter(lambda x : x != obs, self.observers))

    def notify_observers(self):
        #notify observers of current state
        for obs in self.observers:
            obs.notify(self.state)

    def set_state(self, state):
        #set the current state and notify observers of change
        self.state = state
        self.notify_observers()
